zebrad/tests/acceptance.rs

```diff
@@ -174,6 +184,19 @@ macro_rules! assert_with_context {
             panic!("Error: {:?}", report);
         }
     };
+    ($pred:expr, $source:expr, $($fmt_arg:tt)+) => {
+        if !$pred {
+            use color_eyre::Section as _;
+            use color_eyre::SectionExt as _;
+            use zebra_test::command::ContextFrom as _;
+            let report = color_eyre::eyre::eyre!("failed assertion")
+                .section(stringify!($pred).header("Predicate:"))
+                .context_from($source)
+                .wrap_err(format!($($fmt_arg)+));
+
+            panic!("Error: {:?}", report);
+        }
+    };
```

Comment:
:chefs-kiss:

Reply:
So it turns out Rust macros really aren't that hard at all, once you have a few good examples...